lib/PuppeteerSharp/Page.cs

```diff
@@ -1740,7 +1740,7 @@ await Task.WhenAll(Frames.Select(frame => frame.EvaluateExpressionAsync(expressi
 
             if (response.errorText != null)
             {
-                throw new NavigationException(response.errorText.ToString());
+                throw new NavigationException($"{response.errorText.ToString()} at {url}");
```

Comment:
We could add an `Url` property on the `NavigationException` class

Reply:
Would you leave the url on the `Message` as well?

Reply:
I think so, similar to how the [ArgumentException](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/ArgumentException.cs#L77) works